terraform/cloudfront.tf

```diff
@@ -0,0 +1,79 @@
+locals {
+  has_web_domain_hostnames = length(var.web_domain_hostnames) > 0
+}
+
+resource "aws_cloudfront_distribution" "website" {
+  provider = aws.us-east-1
+
+  origin {
+    domain_name = aws_s3_bucket_website_configuration.website.website_endpoint
+    origin_id   = aws_s3_bucket.website.bucket_regional_domain_name
+
+    custom_origin_config {
+      http_port                = 80
+      https_port               = 443
+      origin_keepalive_timeout = 5
+      origin_protocol_policy   = "http-only"
+      origin_read_timeout      = 30
+      origin_ssl_protocols = [
+        "TLSv1.2"
+      ]
+    }
+  }
+
+  enabled     = true
+  price_class = "PriceClass_All"
+  aliases     = [for url in var.web_domain_hostnames : url]
+  web_acl_id  = aws_wafv2_web_acl.website.arn
+
+  is_ipv6_enabled = true
+
+  default_cache_behavior {
+    allowed_methods = ["GET", "HEAD"]
+    cached_methods  = ["GET", "HEAD"]
+
+    cache_policy_id  = "658327ea-f89d-4fab-a63d-7e88639e58f6" # CachingOptimized
```

Comment:
What is the hash value for `cache_policy_id` ?

Reply:
The guid is needed as this is what the CloudFront API expects to set the `CachingOptimized` managed caching policy. I've put a comment beside the value indicating the type of caching policy.

There's some further info on this here:
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-cache-policies.html#managed-cache-caching-optimized